doc/decisions/map_preloading.md

```diff
@@ -0,0 +1,51 @@
+# Map Preloading
+
+## Problem
+
+The application's main component is the map, which consists of multiple layers and even more objects. However, not all layers are visible simultaneously, nor are they required during the initial loading of the application. Loading all the data at once during the initial load may result in slower performance, therefore a strategy must be developed to preload data on the map to optimize its performance.
+
+## Constraints
+
+1.
+
+## Assumptions
+
+1.
+
+## Solutions
+
+### Full fetch
+
+Load all the data from the backend on the initial load. This will result in a slower initial load, but the map will be ready to use immediately.
+
+### Pagination
+
+Since the canvas data is stored in a nested manner, we cannot simply fetch the data in chunks as we would do it with a flat array structure.
+
+### SQL VIEW tables
+
+The data can be preloaded on the database side by creating a view that will contain only the data required for the initial load.
```

Comment:
Would this also be feasible if we want the layers activated/deactivated like they were last time for the user? (Create new SQL VIEWs dynamically)